contracts/src/v0.8/dev/functions/FunctionsOracle.sol

```diff
@@ -2,15 +2,21 @@
 pragma solidity ^0.8.6;
 
 import "../interfaces/FunctionsOracleInterface.sol";
-import "../ocr2/OCR2Base.sol";
-import "./AuthorizedOriginReceiver.sol";
+import "../ocr2/OCR2BaseUpgradeable.sol";
+import "./AuthorizedOriginReceiverUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 
 /**
  * @title Functions Oracle contract
  * @notice Contract that nodes of a Decentralized Oracle Network (DON) interact with
  * @dev THIS CONTRACT HAS NOT GONE THROUGH ANY SECURITY REVIEW. DO NOT USE IN PROD.
  */
-contract FunctionsOracle is FunctionsOracleInterface, OCR2Base, AuthorizedOriginReceiver {
+contract FunctionsOracle is
```

Comment:
Reorder risk again. We must ensure that future upgrades do not reorder these, or if they extend the previous version, must extend like this:

```
contract FunctionOracle2 is FunctionsOracle, SomethingNew { ... }
```

and not:

```
contract FunctionOracle2 is SomethingNew, FunctionsOracle { ... }
```

How can we enforce? 🤔

Reply:
I believe the Hardhat Upgrades plugin enforces storage clashes, even through dependencies, but lemme run a few quick tests to be sure.